storage/interceptors/broker_public_plans_interceptor.go

```diff
@@ -214,13 +225,17 @@ func resyncPlanVisibilitiesWithSupportedPlatforms(ctx context.Context, txStorage
 }
 
 // platformsAnyMatchesVisibility checks whether any of the platform IDs matches the provided visibility
-func platformsAnyMatchesVisibility(platformIDs []string, visibility *types.Visibility) (int, bool) {
-	for i, platformID := range platformIDs {
-		if visibility.PlatformID == platformID {
-			return i, true
+func platformsAnyMatchesVisibility(supportedPlatforms map[string]*types.Platform, visibility *types.Visibility, tenantKey string) (string, bool, bool) {
```

Comment:
this function is not so clear, wouldn't it be more readable to have this one return the matching platform object, and then pass it to another function that checks if it is tenant scoped?